src/modm/platform/core/cortex/build_id.cpp

```diff
@@ -15,19 +15,18 @@ typedef struct {
     uint32_t namesz;
     uint32_t descsz;
     uint32_t type;
-    uint8_t data[];
+    uint8_t data;
 } ElfNoteSection_t;
-extern "C" const ElfNoteSection_t __build_id[];
+extern "C" const ElfNoteSection_t __build_id;
 
 namespace modm
 {
 
 const std::array<uint8_t, 20>&
 build_id()
 {
-	const uint8_t *const sha1 = &__build_id->data[__build_id->namesz];
-	auto *const hash = reinterpret_cast<const std::array<uint8_t, 20> *>(sha1);
-	return *hash;
+	return *reinterpret_cast<const std::array<uint8_t, 20> *>(
+        &__build_id.data + __build_id.namesz);
```

Comment:
This is also undefined behaviour before and after your change. Casting arbitrary memory to `std::array` where no `std::array` has been created is a [strict aliasing violation](https://en.cppreference.com/w/cpp/language/reinterpret_cast#Type_aliasing) and therefore illegal. A conforming version would be returning a value of `std::span<const uint8_t, 20>` referring to the `uint8_t` data.

Reply:
Sorry, but span do non support const, in cannot compile. I have tried it once.

Reply:
Hm, interesting. It works here: https://gcc.godbolt.org/z/5YEjvc

Reply:
This is maybe the best solution with build-in types, It will break signedness but it's not a disaster for log output.  
` char data[30];`
`return std::string_view (&__build_id.data[__build_id.namesz], &__build_id.data[__build_id.namesz + 20]);`

Reply:
thanks, never noticed the compile time length.